### ISSUE TITLE
Check for QSECOFR before starting the debug service

### ISSUE THREAD

**Issue by worksofliam** [codefori/vscode-ibmi#1782]
Since otherwise it can't be started

**Comment by krethan** [comment]
@worksofliam Why check for QSECOFR? It is a system supplier security user profile.

**Comment by worksofliam** [comment]
@krethan i was told by the debug team that this is a requirement for it to start correctly.

**Comment by krethan** [comment]
Issue I see with this one is that QSECOFR profile may exist but the user signed onto the IBM i may not be authorized to the user profile.

**Comment by worksofliam** [comment]
@worksofliam the user can have QSECOFR authority which is what I believe we need to check for.

@mkwan01 can you confirm this?

**Comment by mkwan01** [comment]
@worksofliam I will try to run more testing in this area before giving you a definite answer. We already know that users without QSECOFR authority can start debug service right now. If the user who starts debug service is also the only debug user, then there is no problem.  Right now we are uncertain on the following question:
If the user who starts debug service has no QSECOFR authority, can other users also use the debugger without any problem?

The QB5 host debug server job requires QSECOFR authority. I am not sure whether the debug service also has this requirement. I will do more testing to find it out.

**Comment by mkwan01** [comment]
I have tested using user profiles in the *USER and *PGMR classes. They are able to start debug service and other users on the same IBM i can debug. The only requirement is that they need to have *USE authority to the following programs:
QSYS/STRDBG
QSYS/ENDDBG
QSYS/STRSRVJOB
QSYS/ENDSRVJOB

Not having QSECOFR authority has one impact on the user: he is not able to create the keystore certificates under the /QIBM/ProdData/IBMiDebugService/bin/certs path. You can close this issue if you already have another issue to address the keystore location problem.

**Comment by worksofliam** [comment]
@sebjulliand I believe this can be closed as of 2.10.0?

**Comment by sebjulliand** [comment]
Yup; #1990 fixed this.
